host-configs/rzwhippet-toss_4_x86_64_ib-clang@14.0.6.cmake

```diff
@@ -0,0 +1,112 @@
+
+set(CMAKE_BUILD_TYPE "Release" CACHE STRING "")
+
+#------------------------------------------------------------------------------
+# Compilers
+#------------------------------------------------------------------------------
+# Compiler Spec: clang@=14.0.6
+#------------------------------------------------------------------------------
+if(DEFINED ENV{SPACK_CC})
+
+  set(CMAKE_C_COMPILER "/usr/WS1/axom/libs/toss_4_x86_64_ib/2023_07_13_22_05_10/spack/lib/spack/env/clang/clang" CACHE PATH "")
+
+  set(CMAKE_CXX_COMPILER "/usr/WS1/axom/libs/toss_4_x86_64_ib/2023_07_13_22_05_10/spack/lib/spack/env/clang/clang++" CACHE PATH "")
+
+  set(CMAKE_Fortran_COMPILER "/usr/WS1/axom/libs/toss_4_x86_64_ib/2023_07_13_22_05_10/spack/lib/spack/env/clang/gfortran" CACHE PATH "")
+
+else()
+
+  set(CMAKE_C_COMPILER "/usr/tce/packages/clang/clang-14.0.6/bin/clang" CACHE PATH "")
+
+  set(CMAKE_CXX_COMPILER "/usr/tce/packages/clang/clang-14.0.6/bin/clang++" CACHE PATH "")
+
+  set(CMAKE_Fortran_COMPILER "/usr/tce/packages/gcc/gcc-10.3.1/bin/gfortran" CACHE PATH "")
+
+endif()
+
+set(CMAKE_GENERATOR "Unix Makefiles" CACHE STRING "")
+
+set(ENABLE_FORTRAN ON CACHE BOOL "")
+
+set(BLT_EXE_LINKER_FLAGS " -Wl,-rpath,/usr/tce/packages/clang/clang-14.0.6/lib" CACHE STRING "Adds a missing libstdc++ rpath")
+
+#------------------------------------------------------------------------------
+# MPI
+#------------------------------------------------------------------------------
+
+set(MPI_C_COMPILER "/usr/tce/packages/mvapich2/mvapich2-2.3.6-clang-14.0.6/bin/mpicc" CACHE PATH "")
+
+set(MPI_CXX_COMPILER "/usr/tce/packages/mvapich2/mvapich2-2.3.6-clang-14.0.6/bin/mpicxx" CACHE PATH "")
+
+set(MPI_Fortran_COMPILER "/usr/tce/packages/mvapich2/mvapich2-2.3.6-clang-14.0.6/bin/mpif90" CACHE PATH "")
+
+set(MPIEXEC_EXECUTABLE "/usr/bin/srun" CACHE PATH "")
+
+set(MPIEXEC_NUMPROC_FLAG "-n" CACHE STRING "")
+
+set(ENABLE_MPI ON CACHE BOOL "")
+
+#------------------------------------------------------------------------------
+# Hardware
+#------------------------------------------------------------------------------
+
+#------------------------------------------------
+# Hardware Specifics
+#------------------------------------------------
+
+set(ENABLE_OPENMP ON CACHE BOOL "")
+
+set(ENABLE_GTEST_DEATH_TESTS ON CACHE BOOL "")
+
+#------------------------------------------------------------------------------
+# TPLs
+#------------------------------------------------------------------------------
+
+set(TPL_ROOT "/usr/WS1/axom/libs/toss_4_x86_64_ib/2023_07_13_22_05_10/clang-14.0.6" CACHE PATH "")
+
+set(CONDUIT_DIR "${TPL_ROOT}/conduit-0.8.8-inqrhwboacvngevqbvavhpisx4aeqpvy" CACHE PATH "")
+
+set(C2C_DIR "${TPL_ROOT}/c2c-1.8.0-5smd5ktj7yjc2egeyum4t5vlwmw3jktt" CACHE PATH "")
+
+set(MFEM_DIR "${TPL_ROOT}/mfem-4.5.2-lwet4c3edrdvipij3r4itukmse2jklvy" CACHE PATH "")
+
+set(HDF5_DIR "${TPL_ROOT}/hdf5-1.8.22-5mfm2r7fo5krpj7vvmayz24qksnluryl" CACHE PATH "")
+
+set(LUA_DIR "/usr" CACHE PATH "")
```

Comment:
The only surprise was that it picked up the system lua (5.3, shared) rather than one built with spack

Reply:
Mystery solved: 
https://github.com/LLNL/axom/blob/076803fafe41dcedd963507ea7dc31fc77a2b040/scripts/spack/configs/toss_4_x86_64_ib/spack.yaml#L198-L202

Is this what we want?